src/snowflake/snowpark/dataframe.py

```diff
@@ -4069,7 +4068,7 @@ def _show_string(self, n: int = 10, max_width: int = 50, **kwargs) -> str:
             # Phase 0 code where string gets formatted.
             if is_sql_select_statement(query):
                 result, meta = self._session._conn.get_result_and_metadata(
-                    self.limit(n)._plan, **kwargs
```

Comment:
fixes for show part of relational_grouped_dataframe PR: https://github.com/snowflakedb/snowpark-python/pull/2187